apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNode.tsx

```diff
@@ -119,7 +131,22 @@ export function WorkflowNode({
     }
 
     return true;
-  }, [integrations, tabKey, isEmailLimitReached, isSmsLimitReached]);
+  }, [integrations, tabKey, isEmailLimitReached, isSmsLimitReached, isChannelStep]);
+
+  const getPrimaryIntegration = useMemo(() => {
+    if (channelType === StepTypeEnum.IN_APP) {
+      return InAppProviderIdEnum.Novu;
+    }
+    if (isChannelStep && hasActiveIntegration) {
+      if ([StepTypeEnum.CHAT, StepTypeEnum.PUSH].includes(channelType)) {
+        return integrations?.filter((integration) => integration.channel === channelKey)[0].providerId;
```

Comment:
we should also only filter the current env integrations...
the same for the `hasActiveIntegration`... 
I know this code was here before your changes, but let's please update it 🙏